core/calculator.js

```diff
@@ -670,12 +834,28 @@ function generate_chart(edges, node_quantities) {
 	var nodes = {};
 	for (let column_id in columns) {
 		for (let node_id in columns[column_id]) {
-			var node_name = columns[column_id][node_id];
+			let node_name = columns[column_id][node_id];
+
+			let input = get_input_size(edges, node_name);
+			let output;
+			if (node_name.endsWith(inventory_label_suffix)) {
+				output = used_from_inventory[node_name.replace(inventory_label_suffix, "")];
+			}
+			else {
+				output = node_quantities[node_name];
+			}
+
+			if (used_from_inventory[node_name] !== undefined) {
```

Comment:
Will this ever be `!== undefined`? It looks like you populate `used_from_inventory` with the suffix-less name as queried a few lines up.

Reply:
I'd have to look up the details, but from a glimpse this should be `undefined` by default and only be set for inventory items.

`used_from_inventory` should not be manipulated in this function at all.

Reply:
Just tried it in the debugger and yes, every item that is not in the inventory will hit this `if`.

Reply:
Ah I see it is needed to make the height of the bar chart accurate. I feel like there is a cleaner way to do this but it is not worth holding up this pr for it and something that can maybe be looked at down the line.